From 37f54643d9d09400ed57a3ec9c6117275f73508e Mon Sep 17 00:00:00 2001
From: Alexander Jung <a.jung@lancs.ac.uk>
Date: Thu, 15 Nov 2018 17:00:38 +0000
Subject: [PATCH] [PATCH 4/7] Makefile: HyperNF adaptation

---
 xen/Rules.mk        | 3 +++
 xen/common/Makefile | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/xen/Rules.mk b/xen/Rules.mk
index 5337e20..b53b9d0 100644
--- a/xen/Rules.mk
+++ b/xen/Rules.mk
@@ -57,6 +57,9 @@ CFLAGS += -Werror -Wredundant-decls -Wno-pointer-arith
 CFLAGS += -pipe -g -D__XEN__ -include $(BASEDIR)/include/xen/config.h
 CFLAGS += '-D__OBJECT_FILE__="$@"'
 
+NETMAP_SRC ?= $(realpath $(XEN_ROOT)/../netmap)
+CFLAGS += -I$(NETMAP_SRC)/sys -I$(NETMAP_SRC)/LINUX
+
 ifneq ($(clang),y)
 # Clang doesn't understand this command line argument, and doesn't appear to
 # have an suitable alternative.  The resulting compiled binary does function,
diff --git a/xen/common/Makefile b/xen/common/Makefile
index 24d4752..4e36ab1 100644
--- a/xen/common/Makefile
+++ b/xen/common/Makefile
@@ -63,6 +63,9 @@ obj-y += vmap.o
 obj-y += vsprintf.o
 obj-y += wait.o
 obj-bin-y += warning.init.o
+obj-y += xennet/network.o
+obj-y += xennet/vale.o
+obj-y += xennet/i40e/i40e_netmap_linux.o
 obj-$(CONFIG_XENOPROF) += xenoprof.o
 obj-y += xmalloc_tlsf.o
 
-- 
2.7.4

